ui/src/app/base/components/Header/Header.tsx

```diff
@@ -0,0 +1,304 @@
+  logout: () => void;
+  uuid?: string | null;
+  version?: string;
+};
+
+type NavItem = {
+  adminOnly?: boolean;
+  highlight?: string | string[];
+  inHardwareMenu?: boolean;
+  label: string;
+  url: string;
+};
+
+const navLinks: NavItem[] = [
+  {
+    highlight: ["/machine", "/pool", "/tag"],
+    inHardwareMenu: true,
+    label: "Machines",
+    url: "/machines",
+  },
+  {
+    highlight: "/device",
+    inHardwareMenu: true,
+    label: "Devices",
+    url: "/devices",
+  },
+  {
+    adminOnly: true,
+    highlight: "/controller",
+    inHardwareMenu: true,
+    label: "Controllers",
+    url: "/controllers",
+  },
+  {
+    inHardwareMenu: true,
+    label: "KVM",
+    url: "/kvm",
+  },
+  {
+    label: "Images",
+    url: "/images",
+  },
+  {
+    highlight: "/domain",
+    label: "DNS",
+    url: "/domains",
+  },
+  {
+    highlight: "/zone",
+    label: "AZs",
+    url: "/zones",
+  },
+  {
+    highlight: ["/networks", "/subnet", "/space", "/fabric", "/vlan"],
+    label: "Subnets",
+    url: "/networks?by=fabric",
+  },
+  {
+    adminOnly: true,
+    label: "Settings",
+    url: "/settings",
+  },
+];
+
+const generateURL = (
+  url: NavItem["url"],
+  isLegacy: boolean,
+  appendNewBase: boolean
+) => {
+  if (isLegacy) {
+    return generateLegacyURL(url);
+  } else if (appendNewBase) {
+    return generateNewURL(url, appendNewBase);
+  }
+  return url;
+};
+
+const isSelected = (path: string, link: NavItem, appendNewBase: boolean) => {
+  // Use the provided highlight(s) or just use the url.
+  let highlights = link.highlight || link.url;
+  // If the provided highlights aren't an array then make them one so that we
+  // can loop over them.
+  if (!Array.isArray(highlights)) {
+    highlights = [highlights];
+  }
+  // Check if one of the highlight urls matches the current path.
+  return highlights.some(
+    (start) =>
+      // Check the full path, for both legacy/new clients as sometimes the lists
+      // are in one client and the details in the other.
+      path.startsWith(generateURL(start, true, appendNewBase)) ||
+      path.startsWith(generateURL(start, false, appendNewBase))
+  );
+};
+
+const generateItems = (
+  links: NavItem[],
+  location: Props["location"],
+  appendNewBase: boolean,
+  forHardwareMenu: boolean
+) => {
+  if (forHardwareMenu) {
+    // Only include the items for the hardware menu.
+    links = links.filter((link) => link.inHardwareMenu);
+  }
+  const path = location.pathname + location.search;
+  return links.map((link) => ({
+    className: classNames("p-navigation__item", {
+      // Items that are also displayed in the hardware menu need to be hidden
+      // when the hardware menu is visible.
+      "u-hide--hardware-menu-threshold":
+        link.inHardwareMenu && !forHardwareMenu,
+    }),
+    isSelected: isSelected(path, link, appendNewBase),
+    key: link.url,
+    label: link.label,
+    url: link.url,
+  }));
+};
+
+export const Header = ({
```

Comment:
This component has some minor changes to remove the `rootScope` prop and related `useEffect` handlers.